pennylane/templates/layers/basic_entangler.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``BasicEntanglerLayers`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.ops import CNOT, RX
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    _check_shape,
+    _check_no_variable,
+    _check_wires,
+    _check_number_of_layers,
+    _get_shape,
+)
+
+
+@template
+def BasicEntanglerLayers(weights, wires, rotation=None):
```

Comment:
Very annoying comment here (😆) but: what are your thoughts on naming this `BasicEntangler` rather than `BasicEntanglerLayers`?

e.g.,
```python
@qml.qnode(dev)
def circuit(params):
    qml.templates.BasicEntagler(params, wires=[0, 1])
    return [qml.PauliZ(i) for i in range(5)]
```

Reply:
(Agreed! Or if `Layers` stays in the name, `BasicEntanglingLayers` to match `StronglyEntanglingLayers`? )

Reply:
I woudn't even mind `StronglyEntangler`!